ui/packages/catalog-ui-search/src/main/webapp/extension-points/navigator/navigator.presentation.tsx

```diff
@@ -0,0 +1,429 @@
+import * as React from 'react'
+import { FormattedMessage } from 'react-intl'
+import MarionetteRegionContainer from '../../react-component/container/marionette-region-container'
+const SaveView = require('../../component/save/workspaces/workspaces-save.view')
+const UnsavedIndicatorView = require('../../component/unsaved-indicator/workspaces/workspaces-unsaved-indicator.view')
+import styled from '../../react-component/styles/styled-components'
+import {
+  Button,
+  buttonTypeEnum,
+} from '../../react-component/presentation/button'
+
+export type Props = {
+  hasUnavailableSources: boolean
+  isSaved: boolean
+  branding: string
+  product: string
+  recentWorkspace?: any
+  recentMetacard?: any
+  uploadEnabled: boolean
+  isDevelopment: boolean
+  navigateToRoute: (route: string) => void
+}
+
+export const Divider = () => {
+  return <div className="is-divider" />
+}
+
+export const Root = styled<Props, 'div'>('div')`
+  height: 100%;
+  width: 100%;
+  overflow: auto;
+`
+
+export const WorkspacesSave = styled<
+  {
+    isSaved: Props['isSaved']
+  },
+  'div'
+>('div')`
+  position: absolute;
+  right: 0px;
+  top: 0px;
+  height: auto;
+  overflow: hidden;
+  max-width: ${props => (props.isSaved ? '0%' : '100%')};
+  font-size: ${props => props.theme.largeFontSize};
+  line-height: ${props => props.theme.minimumButtonSize};
+  ${props =>
+    props.isSaved
+      ? `
+  transition: max-width 0s linear
+  ${props.theme.multiple(10, props.theme.coreTransitionTime, 's')};
+  opacity: 1;
+  `
+      : ''} > * {
+    padding: ${props => props.theme.largeSpacing};
+  }
+`
+
+export const WorkspacesIndicator = styled.div`
+  padding-left: ${props => props.theme.minimumSpacing};
+  display: inline-block;
+`
+
+export const SourcesIndicator = styled<
+  {
+    hasUnavailableSources: Props['hasUnavailableSources']
```

Comment:
Same here

Reply:
Similar to above, I'd rather pass in exactly what we need in this case.